salt/utils/dns.py

```diff
@@ -1216,5 +1204,67 @@ def parse_resolv(src="/etc/resolv.conf"):
             "search": search,
             "options": options,
         }
-    except IOError:
+    except OSError:
+        return {}
+
+
+def parse_systemd_resolve():
+    """
+    Parse the information from systemd-resolve
+    """
+
+    nameservers = []
+    ip4_nameservers = []
+    ip6_nameservers = []
+    domain = ""
+
+    cmd = "systemd-resolve --status"
+    cmd = __salt__["cmd.run_all"](
+        "{}".format(cmd), python_shell=False, output_loglevel="quiet"
+    )
```

Comment:
Why not:
```
cmd = ["systemd-resolve", "--status"]
    cmd = __salt__["cmd.run_all"](
        "{}".format(cmd), python_shell=True, output_loglevel="quiet"
    )
```

Reply:
i thought we wanted to avoid using `python_shell=True`?

Reply:
Hmm, we do. Doesn't help that we use `python_shell` and subprocess uses `shell`.

Also, what I really wanted to write was
```python
cmd = ["systemd-resolve", "--status"]
cmd = __salt__["cmd.run_all"](cmd, python_shell=False, output_loglevel="quiet")
```

Reply:
Ie, pass the command as a list, not as a string.

Reply:
Since we can use a different, auto generated resolv.conf file don't need this code.